PyQt5-stubs/sip.pyi

```diff
@@ -41,7 +41,7 @@ Buffer = Union['array', 'voidptr', str, bytes, bytearray]
 
 
 # The array type.
-class array(Sequence): ...
+class array(Sequence): ...  # type: ignore[misc]
```

Comment:
```
PyQt5-stubs/sip.pyi:44: error: Class PyQt5-stubs.sip.array has abstract attributes "__getitem__", "__len__"  [misc]
PyQt5-stubs/sip.pyi:44: note: If it is meant to be abstract, add 'abc.ABCMeta' as an explicit metaclass
```

Reply:
Is it possible to do as the error suggests and make `array` abstract?

Reply:
Or to add the missing abstract members to the stub class?

Reply:
So...  I was trying to dig further and `PyQt5.sip.array` doesn't even seem to exist.  I haven't managed to create an instance yet either by trying to call methods supposedly returning one.  I was hoping I could both explore it and figure out where it comes from.  Though for all I know it just gets converted to a `list` anyways...  So I'm not clear how to decide what the best option is here.

```python
>>> from PyQt5 import sip
>>> sip.array
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'PyQt5.sip' has no attribute 'array'
>>> print('\n'.join(dir(sip)))
SIP_VERSION
SIP_VERSION_STR
_C_API
__doc__
__file__
__loader__
__name__
__package__
__spec__
_unpickle_enum
_unpickle_type
assign
cast
delete
dump
enableautoconversion
enableoverflowchecking
getapi
isdeleted
ispycreated
ispyowned
setapi
setdeleted
setdestroyonexit
settracemask
simplewrapper
transferback
transferto
unwrapinstance
voidptr
wrapinstance
wrapper
wrappertype
```

Reply:
![image](https://user-images.githubusercontent.com/5437880/94353400-a19ad180-0025-11eb-8f7a-6577c472e7fa.png)
I found this in the SIP documentation. 

> returns `the sip.array` object

I feel like this is important, but I know next to nothing about SIP:
> `sip.array` objects are not supported by the `sip5` code generator. They can only be created by handwritten code or by `sip.voidptr.asarray()`.

Reply:
Makes it sound like a non-abstract thing...  maybe?  It takes a `len` parameter so you'd think you could `len()` it which would mean it'd have `.__len__()`.  Probably?  And an array isn't much of an array without being able to index it so `.__getitem__()` seems reasonable.  I think I'm on board with adding those.  Just a sec...

Reply:
Ok I was able to make a `sip.array` object using [this](https://stackoverflow.com/questions/51169641/pythonic-way-to-copy-list-to-a-sip-voidptr-from-qsharedmemory) as reference:
```python
import struct
from PyQt5 import QtCore

shared_mem = QtCore.QSharedMemory()
shared_mem.setNativeKey("test")
shared_mem.create(4*6)
shared_mem.attach()

memtocopy = [0,1,2,3,4,5]

shared_mem.lock()
shared_mem.data()[:] = memoryview(struct.pack('=6i', *memtocopy))
shared_mem.unlock()

array = shared_mem.data().asarray()
print(type(array))
```
```
<class 'sip.array'>
```

Reply:
```pycon
>>> print("".join(map(str, array)))
000010002000300040005000
```
Very strange

Reply:
I think the reason mypy complains that it should be abstract is because `collections.abc.Sequence` is abstract

Reply:
```pycon
>>> dir(array)
['__class__', '__delattr__', '__delitem__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getitem__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__len__', '__lt__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__setitem__', '__sizeof__', '__str__', '__subclasshook__']
```